zebra-chain/src/orchard/shielded_data.rs

```diff
@@ -38,6 +38,29 @@ pub struct ShieldedData {
     pub binding_sig: Signature<Binding>,
 }
 
+impl ShieldedData {
+    /// Iterate over the [`Action`]s for this transaction, returning them as
+    /// their generic type.
```

Comment:
Unlike sapling, there are no generic types in the iterator item type `&Action`.

Here are some examples of generic types: https://doc.rust-lang.org/rust-by-example/generics.html
```suggestion
    /// Iterate over the [`Action`]s for this transaction.
```